extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/service/bytecodeprovider/RuntimeBytecodeProvider.java

```diff
@@ -35,6 +37,25 @@ public ReflectionOptimizer getReflectionOptimizer(
 
     @Override
     public ReflectionOptimizer getReflectionOptimizer(Class<?> clazz, Map<String, PropertyAccess> propertyAccessMap) {
+        try {
+            Class<?> instantiatorClass = Class.forName(clazz.getName() + INSTANTIATOR_SUFFIX, true,
+                    Thread.currentThread().getContextClassLoader());
+            ReflectionOptimizer.InstantiationOptimizer optimizer = (ReflectionOptimizer.InstantiationOptimizer) instantiatorClass
+                    .getDeclaredConstructor().newInstance();
+            return new ReflectionOptimizer() {
+                @Override
+                public InstantiationOptimizer getInstantiationOptimizer() {
+                    return optimizer;
+                }
+
+                @Override
+                public AccessOptimizer getAccessOptimizer() {
+                    return null;
+                }
+            };
+        } catch (Exception e) {
+            // ignored
```

Comment:
There is a reason, actually: some entities are contributed to Hibernate ORM at bootstrap by integrators (e.g. `hibernate-envers`, depending on configuration) and are not contained in the Jandex Index of managed classes. For these entities, optimizing instantiation will not be possible, but throwing / logging an error in that case would be wrong.

Not sure there is a clear way of distinguishing those from "real" entities, especially since potentially external integrators could also add to Hibernate's mappings.

Reply:
Ah, I see, thanks. This looks like a great explanation to replace that `// ignored` comment? :)

As to distinguising these types from user ones... we do have `io.quarkus.hibernate.orm.deployment.spi.AdditionalJpaModelBuildItem`, which seems to be used by envers to let Quarkus know of its additional entity classes. We even seem to tale them into account when building the `JpaModelIndexBuildItem`, so it's possible your patch actually does generate optimizers for these classes:

https://github.com/quarkusio/quarkus/blob/cca25b70ae5b98b89bca76bf404f88ecb8bf3f48/extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java#L357

Reply:
Indeed, they are iterated through when looking at `JpaModelBuildItem#getManagedClassNames`, but we do not have any `ClassInfo` from the jandex index to check if the class is abstract and if it has a no-args constructor, so we actually _do not_ generate optimizers for them. 

Also, as I said possibly (though I'm not sure) there might be other contributors to the entity classes that are not adding to the index, so that was meant to catch all cases.

> This looks like a great explanation to replace that `// ignored` comment? :)

Of course, sorry should've thought about it. I've updated the comment now, lmk if that's clear enough.

Reply:
> Also, as I said possibly (though I'm not sure) there might be other contributors to the entity classes that are not adding to the index, so that was meant to catch all cases.

Pretty sure those other contributors would need to use the same build item, or their contribution would be ignored, given the customization we have in Quarkus (in particular we override the scanner).

Regarding the Jandex index, you should probably use the index from `JpaModelIndexBuildItem`? It's supposed to contain all relevant classes:

https://github.com/quarkusio/quarkus/blob/cca25b70ae5b98b89bca76bf404f88ecb8bf3f48/extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java#L354-L360

We can still keep the try/catch, mind you. I'm just mentioning all this so that if you optimize things, you optimize them all the way :)

Reply:
> Regarding the Jandex index, you should probably use the index from `JpaModelIndexBuildItem`? It's supposed to contain all relevant classes:

Nice, I had missed that one. That's perfect indeed as it also contains additional classes, with my updated solution we should be generating optimizers for those as well.

> We can still keep the try/catch, mind you. I'm just mentioning all this so that if you optimize things, you optimize them all the way :)

Of course, the try-catch is still going to be needed for abstract entity classes or ones missing a no-args constructor - I've rewritten the comment once again to clear this up.

Anyway I updated the PR to also account for additional mappings - thank you for all the help!